honeybee_grasshopper_core/src/HB Apertures by Ratio.py

```diff
@@ -0,0 +1,209 @@
+            different heights will be assigned based on cardinal direction, starting
+            with north and moving clockwise. Default: 0.8 meters.
+        _horiz_separ_: A number for the horizontal separation between
+            individual aperture centerlines.  If this number is larger than
+            the parent face's length, only one aperture will be produced.
+            If an array of values are input here, different separation distances
+            will be assigned based on cardinal direction, starting with north
+            and moving clockwise. Default: 3 meters.
+        vert_separ_: An optional number to create a single vertical
+            separation between top and bottom apertures. If an array of values
+            are input here, different separation distances will be assigned based
+            on cardinal direction, starting with north and moving clockwise.
+            Default: 0.
+        operable_: An optional boolean to note whether the generated Apertures
+            can be opened for ventilation. If an array of booleans are input
+            here, different operable properties will be assigned based on
+            cardinal direction, starting with north and moving clockwise.
+            Default: False.
+        ep_constr_: Optional text for an energy construction to be used for all
+            generated apertures. This text will be used to look up a construction
+            in the window construction library. This can also be a custom
+            WindowConstruction object. If an array of text or construction objects
+            are input here, different constructions will be assigned based on
+            cardinal direction, starting with north and moving clockwise.
+        rad_mat_: Optional text for a radiance material to be used for all
+            generated apertures. This text will be used to look up a material
+            in the material library. This can also be a custom material object.
+            If an array of text or material objects are input here, different
+            materials will be assigned based on cardinal direction, starting
+            with north and moving clockwise.
+        _run: Set to True to run the component.
+    
+    Returns:
+        report: Reports, errors, warnings, etc.
+        hb_obj: The input Honeybee Face or Room with Apertures generated from
+            the input parameters.
+"""
+
+ghenv.Component.Name = "HB Apertures by Ratio"
+ghenv.Component.NickName = 'AperturesByRatio'
+ghenv.Component.Message = '0.1.0\nOCT_28_2019'
+ghenv.Component.Category = "HoneybeeCore"
+ghenv.Component.SubCategory = '0 :: Create'
+ghenv.Component.AdditionalHelpFromDocStrings = "4"
+
+try:  # import the core honeybee dependencies
+    from honeybee.boundarycondition import Outdoors
+    from honeybee.facetype import Wall
+    from honeybee.room import Room
+    from honeybee.face import Face
+    from ladybug_rhino.config import tolerance, conversion_factor
```

Comment:
I would need to know what grouping of dependencies you are thinking of separating into their own error message, then.  `ladybug_rhino` is a dependency of `honeybee-grasshopper-core` so that's why the message says it failed to import honeybee.  Right now, I'm only separating out the import errors for extensions and not dependencies so, if you want separate errors for different dependencies, we need some rules about which dependencies are important enough to get a different message.

It's also worth highlighting that exception the component raises right now includes the more explicit message about the lack of  `ladybug_rhino`.

Reply:
```python
try:  # import the core honeybee dependencies
    from honeybee.boundarycondition import Outdoors
    from honeybee.facetype import Wall
    from honeybee.room import Room
    from honeybee.face import Face
except ImportError as e:
    raise ImportError('\nFailed to import honeybee:\n\t{}'.format(e))

try:
    from ladybug_rhino.config import tolerance, conversion_factor
    from ladybug_rhino.grasshopper import all_required_inputs
except ImportError as e:
    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
```

Reply:
That still doesn't answer my question. Which ones of these dependencies should be grouped together in the same try/except  (since these are all dependencies of honeybee-energy-core):
* ladybug
* ladybug-geometry
* ladybug-dotnet
* ladybug-rhino
* honeybee-core

If they should all have their own try/excepts, then just say so.

Reply:
I'm not sure what this has to do with the dependencies. When you import from honeybee it should give an error that it failed to import honeybee. When you import ladybug_rhino the error should be that it failed to import ladybug_rhino. As simple as that. For other cases, if you are importing from a different library then I think the error should reference to that library.